### PR TITLE
[Link] iteration

### LINKED ISSUE #8794: [Button and Link quality] [Milestone 1] Update `Link` component

### PR DESCRIPTION
**Pull request by aveline**
### WHY are these changes introduced?

Fixes #8794 #8797

### WHAT is this pull request doing?

- Removes `BannerContext` from `Link` component
- Removed `external` prop from `Link` component

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by chazdean — APPROVED** [review]
Changes look good to me! I like the CSS approach in Banner ✨

File: polaris-react/src/components/Link/Link.tsx [hunk 1]
```diff
@@ -14,8 +13,6 @@ export interface LinkProps {
   url?: string;
   /** The content to display inside the link */
   children?: React.ReactNode;
-  /** Makes the link open in a new tab */
-  external?: boolean;
```
Comment (chloerice) [inline polaris-react/src/components/Link/Link.tsx:18]: This prop will need to be deprecated first and then we can remove it in v12.

```suggestion
  /**
    ** @deprecated Set `target` to "_blank" instead
    ** Makes the link open in a new tab
  */
  external?: boolean;
```

**Review by chloerice — COMMENTED** [review]
Looks great @aveline! We'll need to revert removing the `external` prop and deprecate instead.

**Comment by aveline** [comment]
Closing for https://github.com/Shopify/polaris/pull/9223

**Pull request closed without merging**